CoreCompat.NLog.Mongo/MongoTarget.cs

```diff
@@ -328,15 +339,12 @@ private IMongoCollection<BsonDocument> GetCollection()
 
             return _collectionCache.GetOrAdd(key, k =>
             {
-                // create collection
-                var mongoUrl = new MongoUrl(ConnectionString);
-                var client = new MongoClient(mongoUrl);
+                var client = GetMongoClient();
```

Comment:
My intention was to keep the behavior exactly the same when there is no MongoClientSettings provided.

Reply:
One tiny suggestion. Every time I see method without parameters, it is usually a great candidate for Lazy Initialized property or just Property. Look how cleaner the use of it will be:
```
var databaseName = GetDatabaseName();
var database = client.GetDatabase(databaseName);
```
vs
`var database = client.GetDatabase(DatabaseName);`

Reply:
Thanks for the suggestion. I agree properties makes it cleaner. I did originally consider changing the get on existing DatabaseName property, but decided against it since I didn't want to change how any of the public properties worked to avoid making a breaking change.  I moved both private functions to private properties though. Let me know what you think.